### ISSUE TITLE
Some package versions trick `yarn outdated`

### ISSUE THREAD

**Issue by esphen** [yarnpkg/yarn#1510]
**Do you want to request a _feature_ or report a _bug_?**
bug

**What is the current behavior?**
When running `yarn outdated`, some packages are listed, even though they are up to date.

**If the current behavior is a bug, please provide the steps to reproduce.**

``` bash
thor :: ~ % mkdir test
thor :: ~ % cd test
thor :: ~/test % yarn add html5
yarn add v0.16.1
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
success Saved 89 new dependencies.
├─ ansi-regex@2.0.0
├─ ansi-styles@2.2.1
├─ asn1@0.2.3
├─ assert-plus@0.2.0
├─ asynckit@0.4.0
├─ aws-sign2@0.6.0
├─ aws4@1.5.0
├─ bcrypt-pbkdf@1.0.0
├─ bindings@1.2.1
├─ boom@2.10.1
├─ buffer-shims@1.0.0
├─ caseless@0.11.0
├─ chalk@1.1.3
├─ combined-stream@1.0.5
├─ commander@2.9.0
├─ contextify@0.1.15
├─ core-util-is@1.0.2
├─ cryptiles@2.0.5
├─ cssom@0.3.1
├─ cssstyle@0.2.37
├─ dashdash@1.14.0
│  └─ assert-plus@1.0.0
├─ delayed-stream@1.0.0
├─ dom-serializer@0.1.0
│  └─ domelementtype@1.1.3
├─ domelementtype@1.3.0
├─ domhandler@2.3.0
├─ domutils@1.5.1
├─ ecc-jsbn@0.1.1
├─ entities@1.1.1
├─ escape-string-regexp@1.0.5
├─ extend@3.0.0
├─ extsprintf@1.0.2
├─ forever-agent@0.6.1
├─ form-data@2.1.1
├─ generate-function@2.0.0
├─ generate-object-property@1.2.0
├─ getpass@0.1.6
│  └─ assert-plus@1.0.0
├─ graceful-readlink@1.0.1
├─ har-validator@2.0.6
├─ has-ansi@2.0.0
├─ hawk@3.1.3
├─ hoek@2.16.3
├─ html5-entities@1.0.0
├─ html5@v1.0.5
├─ htmlparser2@3.9.2
├─ http-signature@1.1.1
├─ inherits@2.0.3
├─ is-my-json-valid@2.15.0
├─ is-property@1.0.2
├─ is-typedarray@1.0.0
├─ isarray@1.0.0
├─ isstream@0.1.2
├─ jodid25519@1.0.2
├─ jsbn@0.1.0
├─ jsdom@0.11.1
├─ json-schema@0.2.3
├─ json-stringify-safe@5.0.1
├─ jsonpointer@4.0.0
├─ jsprim@1.3.1
├─ mime-db@1.24.0
├─ mime-types@2.1.12
├─ nan@2.4.0
├─ node-uuid@1.4.7
├─ nwmatcher@1.3.9
├─ oauth-sign@0.8.2
├─ opts@1.2.2
├─ pinkie-promise@2.0.1
├─ pinkie@2.0.4
├─ process-nextick-args@1.0.7
├─ punycode@1.4.1
├─ qs@6.3.0
├─ readable-stream@2.1.5
├─ request@2.76.0
├─ sntp@1.0.9
├─ sshpk@1.10.1
│  └─ assert-plus@1.0.0
├─ string_decoder@0.10.31
├─ stringstream@0.0.5
├─ strip-ansi@3.0.1
├─ supports-color@2.0.0
├─ tough-cookie@2.3.2
├─ tunnel-agent@0.4.3
├─ tweetnacl@0.14.3
├─ util-deprecate@1.0.2
├─ verror@1.3.6
├─ xmlhttprequest@1.8.0
└─ xtend@4.0.1
Done in 1.92s.
thor :: ~/test % yarn outdated
yarn outdated v0.16.1
warning No license field
Package Current Wanted Latest
html5   v1.0.5  1.0.5  1.0.5 
Done in 0.14s
```

**What is the expected behavior?**
`yarn outdated` does not show a package with the newest version as outdated.

**Please mention your node.js, yarn and operating system version.**

```
thor :: ~ % yarn --version
0.16.1
thor :: ~ % node --version
v6.4.0
thor :: ~ % uname -a
Linux thor.osl.manamind.com 4.7.9-200.fc24.x86_64 #1 SMP Thu Oct 20 14:26:16 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by watilde** [comment]
I've confirmed this issue with the master branch. There were also a strange behavior like the following. The version number in the package.json is having the `v` prefix unintentionally.

    npm init -y && yarn add html5 && cat package.json

```
{
  "name": "foo",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "html5": "^v1.0.5"
  }
}
```

**Comment by wyze** [comment]
Fixed via #1696.
